### PR TITLE
8274039: codestrings gtest fails when hsdis is present

### PR DESCRIPTION
**Pull request by phedlin**
Please review this change to the (g)test-case "codestrings".

Adding missing regexp pattern to remove trailing **hsdis** printout due to padding on x64.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274039](https://bugs.openjdk.java.net/browse/JDK-8274039): codestrings gtest fails when hsdis is present

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 597e779b1e0db5b8cf2c1c9bd1e7d832694434e7

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5606/head:pull/5606` \
`$ git checkout pull/5606`

Update a local copy of the PR: \
`$ git checkout pull/5606` \
`$ git pull https://git.openjdk.java.net/jdk pull/5606/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5606`

View PR using the GUI difftool: \
`$ git pr show -t 5606`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5606.diff">https://git.openjdk.java.net/jdk/pull/5606.diff</a>

</details>

### CODE REVIEW

**Comment by phedlin** [comment]
Disassembly diff is only used/attempted when **hsdis** is available, e.g. through LD_LIBRARY_PATH, but the approach is rather brittle. Same issue should be expected on other platforms (in particular platforms I do not have access to).

Should the test-case **ifdef** on known platforms?

File: test/hotspot/gtest/code/test_codestrings.cpp [hunk 1]
```diff
@@ -37,10 +37,13 @@ static const char* replace_addr_expr(const char* str)
...
-    // buffer.
+    // buffer (too brittle?).
 
-    std::basic_string<char> tmp = std::regex_replace(str, std::regex("0x[0-9a-fA-F]+"), "<addr>");
-    std::basic_string<char> red = std::regex_replace(tmp, std::regex("\\s+<addr>:\\s+\\.inst\\t<addr> ; undefined"), "");
+    std::basic_string<char> tmp1 = std::regex_replace(str, std::regex("0x[0-9a-fA-F]+"), "<addr>");
+    // Padding: aarch64
+    std::basic_string<char> tmp2 = std::regex_replace(tmp1, std::regex("\\s+<addr>:\\s+\\.inst\\t<addr> ; undefined"), "");
+    // Padding: x64
+    std::basic_string<char> red  = std::regex_replace(tmp2, std::regex("  <addr>:   hlt    \\n"), "");
```
Comment (shipilev) [inline test/hotspot/gtest/code/test_codestrings.cpp:46]: Can we do `\\s+` here as well, like in `aarch64` case? Pretty sure this whitespace can be subtly different.

**Review by shipilev — APPROVED** [review]
This fixes the test failure for my x86_64. Since this is a test bug, I think it is fine to add more platform-specific fixes as needed. 

Alternatively, we might want to compare that the disassembly *prefixes* are the same?

**Review by shipilev — APPROVED** [review]
Not sure why the first group is `[ \\t]+` in x86_64 match, but it still works on my machine.

**Comment by phedlin** [comment]
Just in order not to eat the last "\n" before the padded output.

**Comment by shipilev** [comment]
> Just in order not to eat the last "\n" before the padded output (newlines are consumed at the end of the pattern).
> If you think consuming the newlines in the prefix is better (more obvious or just in line with the aarch64 pattern) we can use:
> 
> `std::regex("\\s+<addr>:\\s+hlt[ \\t]+(?!\\n\\s+;;)")`

This works on my machine too. But honestly, I don't care. Just pick one of these variants and unbreak the `tier1` :)

**Comment by phedlin** [comment]
Thanks for reviewing.
/integrate

**Pull request closed without merging**